Issue/10646 material login flow

khaykov opened the pull request:
Fixes #10646 

This PR moves login flow to the material theme and adds Dark Mode support. 

When deciding on how to approach the styling of the library from the client, I decided to keep the styles overload approach we are using right now. It provides the most flexibility since you are not forced to implement anything. I added some additional styles, so we can control more elements. We are not overriding colors anymore.

Like with other Dark Mode PR's I'm improvising a lot, so feel free to point anything weird. ( eg. I'm not sure what to do with our "greyish" button that we use in a couple of places).

There is a good amount of screens to check, so I split them into couple of sections. I skipped screens that are really similiar:

**Sign-Up**

![comparison_login_signup](https://user-images.githubusercontent.com/728822/73722333-1c039380-46db-11ea-8158-8806344b593d.png)

**Login**

![comparison_login](https://user-images.githubusercontent.com/728822/73722370-2c1b7300-46db-11ea-8ea0-5279eb5e5792.png)

**Landscape layouts**

![comparison_login_landscape](https://user-images.githubusercontent.com/728822/73722402-405f7000-46db-11ea-8fa6-9648f4b8dd86.png)

Notes: 
[This](https://github.com/wordpress-mobile/WordPress-Android/pull/10279#issuecomment-572495028) known issue with new appcompat library is not addressed here.

Username picker/changed is updated in #11112.

To test:
- Go through various login and sign-up flows and make sure everything looks good in both light and dark modes.

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

khaykov commented:
Looks like I missed the landscape version of Signup interstitial layout. 
Thanks to @renanferrari for noticing 👍 

[![Image from Gyazo](https://i.gyazo.com/7d32439eb4dc7428acd6e583e5abac69.png)](https://gyazo.com/7d32439eb4dc7428acd6e583e5abac69)

mattmiklic commented:
> Looks like I missed the landscape version of Signup interstitial layout.

Overall this looks good! I think the filled buttons should be Pink to match the color we're using for them elsewhere.

khaykov commented:
@mattmiklic Updated button style 👍 

[![Image from Gyazo](https://i.gyazo.com/d9be000a747a57e6d7b71667213d75de.png)](https://gyazo.com/d9be000a747a57e6d7b71667213d75de)

![image](https://user-images.githubusercontent.com/728822/74071595-3ed2c800-49b9-11ea-9f69-73484cf82885.png)

jkmassel commented:
We're freezing `14.2` today, so this PR is being bumped to `14.3`. If you'd like it to ship with `14.2`, please merge it into `release/14.2` and ping me – I'll be happy to cut a new beta release.

develric commented on libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/widgets/WPLoginInputRow.java:
```diff
@@ -79,22 +66,8 @@ private void init(Context context, AttributeSet attrs) {
-                    mIcon.setVisibility(View.VISIBLE);
-                } else {
-                    mIcon.setVisibility(View.GONE);
+                    mTextInputLayout.setStartIconDrawable(iconResId);
```
Out of curiosity, I see you "activated" the start icon here. Also I see we removed the `wpIconDrawable` so I was wondering if it was considered to use the start icon instead of totally removing them like below:

![image](https://user-images.githubusercontent.com/47797566/74614568-1f564080-5119-11ea-82af-5d989e4b6aa0.png)

khaykov replied:
Good catch! Yes, we decided to remove the icons after those changes, so it's safe to remove 👍

ashiagr commented:
Tested on Pixel 3a, Android 10 for both light and dark themes changes look good! Great work @khaykov👌

Not sure if it is related to this PR - app crashes on the password screen. I didn't notice it on the `develop` branch. 

Steps to reproduce 
1. Launch app and click Log in
2. Enter Email address and click Next
3. Click Enter your password instead
4. Rotate screen from portrait to landscape and back to portrait

![pwd_crash](https://user-images.githubusercontent.com/1405144/74624106-36bf1900-516d-11ea-9a5f-651e00d407a7.gif)

Crash logs:
```
java.lang.NullPointerException: Attempt to invoke virtual method 'int android.view.View.getVisibility()' on a null object reference
        at org.wordpress.android.login.LoginMagicLinkRequestFragment.onSaveInstanceState(LoginMagicLinkRequestFragment.java:246)
        at androidx.fragment.app.Fragment.performSaveInstanceState(Fragment.java:2766)
        at androidx.fragment.app.FragmentManagerImpl.saveFragmentBasicState(FragmentManagerImpl.java:2289)
        at androidx.fragment.app.FragmentManagerImpl.saveAllState(FragmentManagerImpl.java:2348)
        at androidx.fragment.app.FragmentController.saveAllState(FragmentController.java:151)
        at androidx.fragment.app.FragmentActivity.onSaveInstanceState(FragmentActivity.java:513)
        at androidx.appcompat.app.AppCompatActivity.onSaveInstanceState(AppCompatActivity.java:533)
        at org.wordpress.android.ui.accounts.LoginActivity.onSaveInstanceState(LoginActivity.java:168)

```

khaykov commented:
@ashiagr Thanks! This is a known issue, that we tiptoe around for many months :) Most recent discussion and the fix is in #11292

ashiagr commented:
>  This is a known issue

Sorry didn't realise it earlier! 

On another note, is there any separate PR for styling sub-screens from **Help menu on login/signup screens**? Noticed Application Log screen display logs in black text on a dark background and so they're not properly readable. 

![logs](https://user-images.githubusercontent.com/1405144/74627948-4abd4780-517a-11ea-8e28-66d715892429.png)

khaykov commented:
@ashiagr I totally missed that one! I'll update the color scheme there alongside ZenDesk (hopefully).

@develric Thanks for the review! All legit concerns, so I fixed them :)

Removed the padding:
[![Image from Gyazo](https://i.gyazo.com/af8ff10a026dc7be87a2cfdb7f78df39.png)](https://gyazo.com/af8ff10a026dc7be87a2cfdb7f78df39)

Removed that weird background and using the login theme icon color here:
[![Image from Gyazo](https://i.gyazo.com/afc06b7f250ef42a9c12298b832e53b9.png)](https://gyazo.com/afc06b7f250ef42a9c12298b832e53b9)

guarani commented:
@khaykov I noticed that this PR closes https://github.com/wordpress-mobile/WordPress-Android/issues/10646 – is this correct? The ticket being closed is a dark mode Gutenberg editor integration ticket, which doesn't seem to be what this PR is about.

khaykov commented:
@guarani Yes, it looks like it's targeting the wrong issue.
